lib/Conversion/TTIRToTTIRDecomposition/TTIRToTTIRDecomposition.cpp

```diff
@@ -283,14 +287,149 @@ struct ConvolutionToConv2dPattern
     return success();
   }
 
-  LogicalResult
-  matchAndRewrite(ttir::ConvolutionOp op, OpAdaptor adaptor,
-                  ConversionPatternRewriter &rewriter) const override {
+  ttir::ReshapeOp createReshapeOp(Location loc, Value tensor,
+                                  llvm::ArrayRef<int64_t> target_input_shape,
+                                  ::mlir::ArrayAttr constraints,
+                                  ConversionPatternRewriter &rewriter) const {
+    auto inputType = mlir::cast<RankedTensorType>(tensor.getType());
+
+    auto output = rewriter.create<tensor::EmptyOp>(
+        loc, llvm::ArrayRef<int64_t>(target_input_shape),
+        inputType.getElementType());
+    std::vector<int32_t> shapei32(target_input_shape.begin(),
+                                  target_input_shape.end());
+    auto shape_attr = rewriter.getI32ArrayAttr(shapei32);
+
+    return rewriter.create<ttir::ReshapeOp>(
+        loc,
+        mlir::RankedTensorType::get(target_input_shape,
+                                    inputType.getElementType()),
+        tensor, output, shape_attr, constraints);
+  }
+
+  mlir::DenseI64ArrayAttr
+  addIntegersToDenseArrayAttr(ConversionPatternRewriter &rewriter,
+                              mlir::DenseI64ArrayAttr denseArrayAttr,
+                              std::vector<uint64_t> additionalIntegers) const {
+    llvm::SmallVector<int64_t, 4> newDenseArray(denseArrayAttr.asArrayRef());
+    for (uint64_t integer : additionalIntegers) {
+      newDenseArray.push_back(integer);
+    }
+    return rewriter.getDenseI64ArrayAttr(newDenseArray);
+  }
+
+  mlir::DenseBoolArrayAttr
+  addBooleansToDenseArrayAttr(ConversionPatternRewriter &rewriter,
+                              mlir::DenseBoolArrayAttr denseArrayAttr,
+                              std::vector<bool> additionalIntegers) const {
+    llvm::SmallVector<bool, 4> newDenseArray(denseArrayAttr.asArrayRef());
+    for (bool boolean : additionalIntegers) {
+      newDenseArray.push_back(boolean);
+    }
+    return rewriter.getDenseBoolArrayAttr(newDenseArray);
+  }
 
-    if (failed(isConv2d(op))) {
+  LogicalResult matchAndRewrite1d(ttir::ConvolutionOp op, OpAdaptor adaptor,
```

Comment:
Can you add commentary outlining what's going on in this fn? It's hard to follow thru code only.

Reply:
I have added the comments at the header of a newly added pattern for 1d convolution. Please let me know if you think I should go into more details in the code itself.